### PR TITLE
[RR-158] add session member to raft_entry_t

### PR DESCRIPTION
**Pull request by sjpotter**
enables users of libraft to group raft entries into sessions.

### CODE REVIEW

File: include/raft_types.h [hunk 1]
```diff
@@ -20,6 +20,11 @@ typedef long int raft_term_t;
  */
 typedef long int raft_index_t;
 
+/**
+ * Id used to group entries into sessions
+ */
+typedef unsigned long raft_session_t;
```
Comment (tezc) [inline include/raft_types.h:26]:
```suggestion
typedef unsigned long long raft_session_t;
```
Let's make it 64 bit at least

**Comment by tezc** [comment]
@sjpotter maybe we can add a small test just to touch this part? Probably, you can add a line to one of the existing tests.

**Comment by sjpotter** [comment]
> @sjpotter maybe we can add a small test just to touch this part? Probably, you can add a line to one of the existing tests.

I'm not sure what the test would be.  all this did was add a struct member, no actual running "code".  I was planning on having tests in RedisRaft.

**Comment by sjpotter** [comment]
I think the only test would  be to add session setup/teardown to virtraft (but I think that's beyond the scope of this)

**Review by tezc — APPROVED** [review]

**Pull request merged**